src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,286 @@
+          self.icd_nodeid
+        )
+        logging.info(f"ActiveModeDuration: {active_mode_duration}")
+
+        self.step(4)
+        # Test 1B
+
+        # Subscription to ActiveChanged
+        event = brbinfoCluster.Events.ActiveChanged
+        self.q = queue.Queue()
+        urgent = 1
+        cb = SimpleEventCallback("ActiveChanged", event.cluster_id, event.event_id, self.q)
+        subscription = await self.default_controller.ReadEvent(nodeid=self.dut_node_id, events=[(kICDBridgedEndpointId, event, urgent)], reportInterval=[1, 3])
+        subscription.SetEventUpdateCallback(callback=cb)
+
+        StayActiveDuration = 1000
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration}ms)")
+        self._send_keep_active_command(StayActiveDuration)
+
+        logging.info(f"Waiting for ActiveChanged from DUT...")
+        PromisedActiveDuration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, StayActiveDuration))/1000)
+
+        asserts.assert_true(PromisedActiveDuration >= StayActiveDuration, "PromisedActiveDuration < StayActiveDuration")
+
+        self.step(5)
+        # Test 2
+
+        StayActiveDuration = 1500
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration}ms)")
+        self._send_keep_active_command(StayActiveDuration)
+
+        logging.info(f"Waiting for ActiveChanged from DUT...")
+        PromisedActiveDuration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, StayActiveDuration))/1000)
+
+        # wait for active time duration
+        time.sleep(max(StayActiveDuration/1000, active_mode_duration))
+        # ICD now should be in idle mode
+
+        # sends 3x keep active commands
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration})")
+        self._send_keep_active_command(StayActiveDuration)
+        time.sleep(StayActiveDuration/1000)
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration})")
+        self._send_keep_active_command(StayActiveDuration)
+        time.sleep(StayActiveDuration/1000)
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration})")
+        self._send_keep_active_command(StayActiveDuration)
+        time.sleep(StayActiveDuration/1000)
+
+        logging.info(f"Waiting for ActiveChanged from DUT...")
+        PromisedActiveDuration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, StayActiveDuration))/1000)
+
+        asserts.assert_true(q.qSize() == 0, "More than one event received from DUT")
+
+        self.step(6)
+        # Test 3
+
+        StayActiveDuration = 10000
+        logging.info(f"Sending KeepActiveCommand({StayActiveDuration})")
+        self._send_keep_active_command(StayActiveDuration)
+
+        ## halt the ICD process
+        self.app_process.send_signal(signal.SIGSTOP.value)
+
+        logging.info(f"Waiting for 60 minutes")
```

Comment:
I suggest this last step of waiting 60 mins be skipped if it is CI

```
self.is_ci = self.check_pics('PICS_SDK_CI_ONLY') at the start of the test
```

then check for 
```
if not self.is_ci:
   # Run the 60 min test test step
```

Reply:
ack